pyamf/__init__.py

```diff
@@ -468,19 +497,15 @@ def get_decoder(encoding, *args, **kwargs):
     @type encoding: C{int}
     @raise ValueError: Unknown C{encoding}.
     """
+    use_ext = kwargs.pop('use_ext', None)
+
     def _get_decoder_class():
         if encoding == AMF0:
-            try:
-                from cpyamf import amf0
-            except ImportError:
-                from pyamf import amf0
+            amf0 = _get_amf_module(0, use_ext=use_ext)
```

Comment:
AMF0 == 0 and AMF3 == 3. so you could:

    if encoding in [AMF0, AMF3]: # if encoding hasn't been validated already
        amf_module = _get_amf(encoding, use_ext=use_ext)
        return amf_module.Decoder